backend/core/src/main/java/org/sonarsource/sonarlint/core/spring/SonarLintSpringAppConfig.java

```diff
@@ -261,6 +266,21 @@ MonitoringInitializationParams provideMonitoringInitParams(InitializeParams para
       params.getTelemetryConstantAttributes().getIdeVersion());
   }
 
+  @Bean
+  SonarLintDatabaseInitParams provideStorageInitParams(InitializeParams params) {
+    return new SonarLintDatabaseInitParams(params.getStorageRoot(), SonarLintDatabaseMode.FILE, true);
+  }
+
+  @Bean(destroyMethod = "shutdown")
+  SonarLintDatabase provideSonarLintDatabase(SonarLintDatabaseInitParams sonarLintDatabaseInitParams) {
+    return new SonarLintDatabase(sonarLintDatabaseInitParams);
+  }
+
+  @Bean
+  AiCodeFixRepository provideAiCodeFixRepository(SonarLintDatabase sonarLintDatabase) {
+    return new AiCodeFixRepository(sonarLintDatabase);
+  }
+
```

Comment:
A slightly less verbose way is to declare those 2 beans in the `@Import` annotation at the top of this class